include/zephyr/sys/mem_blocks.h

```diff
@@ -80,6 +80,14 @@ typedef sys_mem_blocks_t *(*sys_multi_mem_blocks_choice_fn_t)
  * @cond INTERNAL_HIDDEN
  */
 
+#ifdef CONFIG_SYS_MEM_BLOCKS_RUNTIME_STATS
+struct sys_mem_blocks_runtime_stats {
```

Comment:
Can't we just use the heap stats struct directly?  (Ditto for slabs too, FWIW).  The allocator behavior is different, but the information collected seems identical.

Reply:
I was originally thinking that it may be prudent to keep them separate in case the info to track diverges. However, the more I think about it, the more I agree that it would be better to have them report using the same structure. If they do diverge, then unused fields could be set to -1, or some other standard value. After all, it is not like there isn't precedence elsewhere--I'm thinking of "struct stat" in particular.

I'm planning on making this change.